defi/src/utils/craftProtocol.ts

```diff
@@ -48,7 +48,13 @@ function selectChainFromItem(item: any, normalizedChain: string) {
   return item[normalizedChain] ?? item[altChainName];
 }
 
-const raisesPromise = fetch("https://api.llama.fi/raises").then((res) => res.json());
+let raisesPromise: Promise<any> | undefined = undefined;
+
+async function getRaises() {
```

Comment:
nice

Reply:
although if this is now a long running process inside a server then we gotta make sure that this data is updated from time to time

Reply:
yeah, it is updated every 24 hours here: https://github.com/DefiLlama/defillama-server/blob/api-v2/defi/src/api2/cache.ts#L115

Reply:
but updateRaises() doesnt change the object returned by getRaises tho?

Reply:
yeah, it will. since raises data is stored in cache object and gets updated: https://github.com/DefiLlama/defillama-server/blob/api-v2/defi/src/api2/cache.ts#L84

Reply:
but the raises that is used to build the protocol info (https://github.com/DefiLlama/defillama-server/pull/4530/files/1b3a8a5ffbe02d5d76bfdf2f03b6309da7011fd1#diff-c29996546052b582d833e7d153f6563c48aeb520c4780298edd25f005b6a5757R224) is not the raises info stored in the cache object but instead its the raises info from getRaises()

raisesPromise/getRaises are independent from cache.raises, arent they?

Reply:
oh yeah, my assumption for craftProtocol was, it is a lambda call, builds response and then exits? unlike api v2, wrong assumption?

Reply:
i think for now is fine, just flagging that if we ever move it to api2 then gotta change that

Reply:
api v2 uses the other one (one from cache module), so data is updated every 24 hours